core/src/main/scala/kafka/coordinator/group/GroupCoordinator.scala

```diff
@@ -416,14 +416,14 @@ class GroupCoordinator(val brokerId: Int,
       case None =>
         // if the group is marked as dead, it means some other thread has just removed the group
         // from the coordinator metadata; it is likely that the group has migrated to some other
-        // coordinator OR the group is in a transient unstable phase. Let the consumer to retry
-        // joining without specified consumer id,
-        responseCallback(Errors.UNKNOWN_MEMBER_ID)
+        // coordinator OR the group is in a transient unstable phase. Let the member retry
+        // joining without resetting specified member id in case we need to fence.
+        responseCallback(Errors.COORDINATOR_NOT_AVAILABLE)
```

Comment:
The comment above doesn't seem to fit the case, perhaps the result of a refactor somewhere along the way. Shall we move it below to the actual `Dead` case?

Reply:
+1